### PR TITLE
Add reject all option to personal sign header

### LINKED ISSUE #22744: Confirmations redesign - implement reject all button for confirmation pages
Currently at bottom of confirmation pages we have feature like below to reject all confirmations:

Design: https://www.figma.com/file/7kYok9B46ohR6h47cksRAz/Confirmation-redesign-V1?type=design&node-id=982-45793&mode=design&t=UXL4JzT6G9DcDYVQ-0

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Confirmation redesign add reject all option to nav bar.

## **Related issues**

Fixes: MetaMask/metamask-extension#22744, MetaMask/metamask-extension#22935

## **Manual testing steps**

1. Enable confirmations redesign locally
2. Submit multiple transactions
3. Check to nav bar on personal sign message page to confirm presence of Reject All button

## **Screenshots/Recordings**
Screenshot 2024-02-14 at 3 35 01 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've clearly explained what problem this PR is solving and how it is solved.
- [X] I've linked related issues
- [X] I've included manual testing steps
- [X] I've included screenshots/recordings if applicable
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [X] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [X] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by segun — APPROVED** [review]

File: ui/pages/confirmations/components/confirm/nav/nav.tsx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    return pendingConfirmations.sort((a1, a2) => a1.time - a2.time).findIndex(
+      ({ id }) => id === currentConfirmation.id,
+    );
+  }, [currentConfirmation, pendingConfirmations]);
+
+  const onNavigateToTransaction = useCallback(
+    (pos: number) => {
+      const nextConfirmation =
+        pendingConfirmations[currentConfirmationPosition + pos];
+      console.log('----- next id ----', pendingConfirmations);
```
Comment (segun) [inline ui/pages/confirmations/components/confirm/nav/nav.tsx:66]:
```suggestion
```

**Review by segun — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js [hunk 2]
```diff
@@ -36,9 +36,12 @@ const ConfirmPageContainerNavigation = () => {
     ...enumUnapprovedEncryptMsgsKey,
   ];
 
-  const enumUnapprovedTxs = Object.keys(unconfirmedTransactions).filter(
-    (key) => enumDecryptAndEncryptMsgs.includes(key) === false,
-  );
+  const enumUnapprovedTxs = Object.keys(unconfirmedTransactions)
```
Comment (cryptotavares) [inline ui/pages/confirmations/components/confirm-page-container/confirm-page-container-navigation/confirm-page-container-navigation.component.js:39]: Why are we sorting here? And if we need sorting can't we use the `unconfirmedTransactionsListSelector` selector (that should already return the unconfirmed transactions sorted)

File: ui/pages/confirmations/hooks/useCurrentConfirmation.ts [hunk 3]
```diff
@@ -34,22 +34,27 @@ const useCurrentConfirmation = () => {
...
       pendingConfirmation = latestPendingConfirmation;
     }
-    if (
-      pendingConfirmation &&
-      pendingConfirmation.id !== currentConfirmation?.id
-    ) {
+    if (!pendingConfirmation) {
+      setCurrentConfirmation(undefined);
+      return;
+    }
```
Comment (cryptotavares) [inline ui/pages/confirmations/hooks/useCurrentConfirmation.ts:44]: What do you feel about the change below?
```js
if (!pendingConfirmation){
  if (!latestPendingConfirmation) {
    setCurrentConfirmation(undefined);
    return;
  }

  pendingConfirmation = latestPendingConfirmation;
}
```

This change would reduce the cyclomatic complexity from 3 to 2.

File: ui/pages/confirmations/components/confirm/nav/nav.tsx [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  const t = useI18nContext();
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+  const pendingConfirmations = useSelector(pendingConfirmationsSelector);
+  const dispatch = useDispatch();
+
+  const currentConfirmationPosition = useMemo(() => {
+    if (pendingConfirmations?.length <= 0 || !currentConfirmation) {
+      return 0;
+    }
+    return pendingConfirmations.sort((a1, a2) => a1.time - a2.time).findIndex(
```
Comment (cryptotavares) [inline ui/pages/confirmations/components/confirm/nav/nav.tsx:57]: shouldn't we move the sort to a selector? Something like `pendingConfirmationsSortedSelector`?

File: ui/pages/confirmations/components/confirm/nav/nav.tsx [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+      ({ id }) => id === currentConfirmation.id,
+    );
+  }, [currentConfirmation, pendingConfirmations]);
+
+  const onNavigateToTransaction = useCallback(
+    (pos: number) => {
+      const nextConfirmation =
+        pendingConfirmations[currentConfirmationPosition + pos];
```
Comment (cryptotavares) [inline ui/pages/confirmations/components/confirm/nav/nav.tsx:65]: How is the next confirmation the one created after/before the current confirmation if pendingConfirmations is not ordered? :thinking: Is it a side effect of sorting the `pendingConfirmations` array when calling `currentConfirmationPosition`? If yes then we should definitely have a selector to get sorted pending confirmations.

File: ui/pages/confirmations/components/confirm/nav/nav.tsx (hunk 1 above)
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/nav/nav.tsx]: Done

File: ui/pages/confirmations/hooks/useCurrentConfirmation.ts (hunk 3 above)
Comment (jpuri) [reply ui/pages/confirmations/hooks/useCurrentConfirmation.ts:44]: looks good, let me make this change

File: ui/pages/confirmations/components/confirm/nav/nav.tsx (hunk 5 above)
Comment (jpuri) [reply ui/pages/confirmations/components/confirm/nav/nav.tsx:65]: I agree, let me make these changes

**Review by pedronfigueiredo — APPROVED** [review]

File: ui/pages/confirmations/components/confirm/nav/nav.test.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  });
+
+  it('renders the "Reject all" Button', () => {
+    const { getAllByRole, getByText } = render();
+    const buttons = getAllByRole('button');
+    expect(buttons).toHaveLength(3);
+    expect(getByText('Reject all')).toBeInTheDocument();
+  });
+
+  it('renders button to navigate to previous or next confirmation', () => {
```
Comment (segun) [inline ui/pages/confirmations/components/confirm/nav/nav.test.tsx:92]: I see that the buttons are disabled when you are at the beginning or end of the list. Can we test this as well.

**Review by segun — APPROVED** [review]
Looks good, left a comment.

**Pull request merged**